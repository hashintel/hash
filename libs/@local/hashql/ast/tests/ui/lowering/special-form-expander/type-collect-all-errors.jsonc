//@ run: fail
//@ description: We should collect all errors when lowering type/3, and not only report the first one.
["::kernel::special_form::type", "::x", { "#literal": 1 }, "x"]
//~^ ERROR Replace this qualified path with a simple identifier
//~| ERROR Replace this literal with a type name
