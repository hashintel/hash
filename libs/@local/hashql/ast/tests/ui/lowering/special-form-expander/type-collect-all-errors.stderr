[31m[special-form-expander::invalid-type-expression] Error:[0m Invalid type expression
   â•­â”€[ <unknown>:3:41 ]
   â”‚
 3 â”‚ ["::kernel::special_form::type", "::x", { "#literal": 1 }, "x"]
   â”‚                                         â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€  
   â”‚                                                 â•°â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ Replace this literal with a type name
   â”‚ 
   â”‚ Help: Replace this expression with a valid type reference, struct type, or tuple type
   â”‚ 
   â”‚ Note: Valid type expressions include:
   â”‚       - Type names: String, Int, Float
   â”‚       - Struct types: {name: String, age: Int}
   â”‚       - Tuple types: (String, Int, Boolean)
   â”‚       - Unions: (| String Int)
   â”‚       - Intersections: (& String Int)
   â”‚       - Generic types: Array<String>, Option<Int>
â”€â”€â”€â•¯


[31m[special-form-expander::qualified-binding-name] Error:[0m Qualified path used as binding name
   â•­â”€[ <unknown>:3:35 ]
   â”‚
 3 â”‚ ["::kernel::special_form::type", "::x", { "#literal": 1 }, "x"]
   â”‚                                   â”€â”¬â”€  
   â”‚                                    â•°â”€â”€â”€ Replace this with a simple identifier
   â”‚ 
   â”‚ Help: type binding names must be simple identifiers without any path qualification. Qualified paths cannot be used as binding names.
   â”‚ 
   â”‚ Note: Valid identifiers are simple names like 'x', 'counter', '+', or 'user_name' without any namespace qualification, generic parameters, or path separators.
â”€â”€â”€â•¯
